File: cs50x/week1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

void print_pyramid(int height);

int main(void)
{
    // Prompt the user for the pyramid's height
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height <= 0 || height > 8);
    print_pyramid(height);
}

void print_pyramid(int height)
{
	for (int i = 1; i <= height; i++)
   {
		for (int j = 1; j <= height; j++)
		{
			if (i + j >= height + 1)
				printf("#");
			else
				printf(" ");
		}
		printf("\n");
	}
}

File: cs50x/week6/sentimental-mario-more/mario.py

from cs50 import get_int

# Prompt user for an integer number between 1 and 8 for the pyramid height
# Manage errors that re-prompt input, for invalid numbers o inputs

while True:
    try:
        height = get_int("Height: ")
        if height >= 1 and height <= 8:
            break

    except ValueError as error:
        print(error)

# Iterate to print two half-pyramids separated for a gap of size 2
for i in range(height):
    print(" " * (height - i - 1), end="")
    print("#" * (i + 1), end="")

    # Print a gap of size 2
    print("  ", end="")

    # Print the second half-pyramid
    print("#" * (i + 1))
